src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -510,12 +511,25 @@ pass::EliminateEltwise::EliminateEltwise() {
     auto constant_pattern = pattern::wrap_type<opset8::Constant>();
     auto eltwise_pattern =
         pattern::wrap_type<opset8::Add, opset8::Subtract, opset8::Multiply, opset8::Divide>({input, constant_pattern});
+    auto subtract_pattern =
+        pattern::wrap_type<opset8::Subtract>({input, pattern::wrap_type<opset8::Convert>({constant_pattern})});
+    auto root = std::make_shared<pattern::op::Or>(OutputVector{eltwise_pattern, subtract_pattern});
 
     matcher_pass_callback callback = [=](pattern::Matcher& m) {
         const auto& pattern_map = m.get_pattern_value_map();
-        auto eltwise = pattern_map.at(eltwise_pattern).get_node_shared_ptr();
-        auto non_const_input = pattern_map.at(input);
-        auto constant = pattern_map.at(constant_pattern);
+
+        std::shared_ptr<Node> eltwise;
+        auto it = pattern_map.find(eltwise_pattern);
+        if (it != pattern_map.end()) {
+            eltwise = it->second.get_node_shared_ptr();
+        } else if ((it = pattern_map.find(subtract_pattern)) != pattern_map.end()) {
+            eltwise = it->second.get_node_shared_ptr();
+        } else {
+            return false;
+        }
```

Comment:
Can we just use m.get_match_root() here to get eltwise node?
```suggestion
        auto eltwise = m.get_match_root();
```

Reply:
done